### DISCUSSION TITLE
Victron MPPT + SH1106 @ ESP32 Wroom 30Pin

### DISCUSSION THREAD

**Discussion by CosmicGMX** [hoylabs/OpenDTU-OnBattery#368] [Q&A] [answered]
Welche Alternativen Pins kann ich für SCL und SDA am ESP32 Wroom mit 30 Pins für die MPPT oder SH1106 Schnittstelle nutzen ?
Für beide sind die GPIO21 und GPIO22 vorgesehen...was aber nicht funktioniert.

**Comment by spcqike** [comment]
Wenn ich die anderen User richtig verstanden hab, kannst du die Pins in deinem eigenen Device Profil frei belegen. I2C würde ich auf 21/22 belassen, da diese controllerintern dafür vorgesehen sind.

Victron RX kannst du quasi frei auf einen verfügbaren Pin legen, der als Input genutzt werden kann (also nicht 0, 1, 3 , 6-11 oder 12) 

Victron TX wird glaube gar nicht verwendet, wenn ich das richtig gelesen hatte ... sonst halt auf einen freien Output PIN.

**Comment by peff74** [comment]
Die gleichen Pins für Display und Victron können nicht funktionieren.
Wo ist denn die so vorgesehen?
Es muss ja in der Pin Mapping hinterlegt werden.
Diese muss dann hochgeladen und aktiviert werden.

**Comment by CosmicGMX** [comment]
Hab's jetzt mit GPIO 34 und 35 für Victron Rx und TX probiert ... PinMapping angepasst... komme jetzt nicht mehr auf's Gerät. -> neu Flashen und andere Pins ausprobieren.

**Reply by peff74** [reply]
Oh, 34/35 ist ne schlechte Wahl gewesen^^ sind input only PINs haben soweit ich weiß nicht mal nen Pullup.
Nimm mal lieber 2/4 12/13 14/15 oder so.

**Reply by Manos1966** [reply]
https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/360#discussioncomment-6690406

**Reply by CosmicGMX** [reply]
Danke  ... jetzt läuft's.

**Comment by CosmicGMX** [comment, marked as answer]
Vielen Dank für Eure Hilfe, jetzt läuft es !
Den Victron habe ich auf GPIO21 / 22 gelassen (inkl. Levershifter). Das Display SH1106 ist jetzt auf GPIO32 / 33 umgezogen. Das ganze läuft jetzt auf einem 30 Pin ESP32 Wroom.
